Mixer e2e TestRateLimit is Flaky

This had happened for quite a number of post submit and release test flakes.
Example: https://prow.istio.io/log?job=e2e-smoke&id=2116

```
I0131 16:59:52.652] --- FAIL: TestRateLimit (120.58s)
I0131 16:59:52.652] 	mixer_test.go:473: Establishing metrics baseline for test...
I0131 16:59:52.652] 	mixer_test.go:475: prometheus query: istio_request_count{destination_service="ratings.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local"}
I0131 16:59:52.652] 	mixer_test.go:483: error getting prior 429s, using 0 as value (msg: value not found for map[string]string{"response_code":"429"})
I0131 16:59:52.652] 	mixer_test.go:492: Baseline established: prior200s = 1.000000, prior429s = 0.000000
I0131 16:59:52.652] 	mixer_test.go:494: Sending traffic...
I0131 16:59:52.653] 	mixer_test.go:526: Fortio Summary: 200 reqs (9.926662 rps, 200.000000 200s (9.926662 rps), 0 400s - map[200:200])
I0131 16:59:52.653] 	mixer_test.go:538: Expected Totals: 200s: 20.147760 (1.000000 rps), 429s: 179.852240 (8.926662 rps)
I0131 16:59:52.653] 	mixer_test.go:549: prometheus query: istio_request_count{destination_service="ratings.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local"}
I0131 16:59:52.653] 	mixer_test.go:568: Actual 429s: 175.000000 (8.685829 rps)
I0131 16:59:52.653] 	mixer_test.go:585: Actual 200s: 24.000000 (1.191199 rps), expecting ~1 rps
I0131 16:59:52.653] 	mixer_test.go:600: prometheus values for istio_request_count:
I0131 16:59:52.653] 		istio_request_count{destination_service="details.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="200", source_service="productpage.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v1"} => 205 @[1517417984.248]
I0131 16:59:52.654] 		istio_request_count{destination_service="details.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="403", source_service="productpage.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v1"} => 1 @[1517417984.248]
I0131 16:59:52.654] 		istio_request_count{destination_service="productpage.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="200", source_service="ingress.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="unknown"} => 206 @[1517417984.248]
I0131 16:59:52.654] 		istio_request_count{destination_service="productpage.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="403", source_service="ingress.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="unknown"} => 1 @[1517417984.248]
I0131 16:59:52.654] 		istio_request_count{destination_service="ratings.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="200", source_service="reviews.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v3"} => 25 @[1517417984.248]
I0131 16:59:52.654] 		istio_request_count{destination_service="ratings.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="429", source_service="reviews.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v3"} => 175 @[1517417984.248]
I0131 16:59:52.655] 		istio_request_count{destination_service="ratings.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v2", instance="10.64.3.228:42422", job="istio-mesh", response_code="200", source_service="reviews.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v3"} => 1 @[1517417984.248]
I0131 16:59:52.655] 		istio_request_count{destination_service="reviews.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="200", source_service="productpage.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v1"} => 4 @[1517417984.248]
I0131 16:59:52.655] 		istio_request_count{destination_service="reviews.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v1", instance="10.64.3.228:42422", job="istio-mesh", response_code="403", source_service="productpage.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v1"} => 2 @[1517417984.248]
I0131 16:59:52.655] 		istio_request_count{destination_service="reviews.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", destination_version="v3", instance="10.64.3.228:42422", job="istio-mesh", response_code="200", source_service="productpage.mixer-test-4b8af8a43fbc441cb9a2de4ae3.svc.cluster.local", source_version="v1"} => 201 @[1517417984.248]
I0131 16:59:52.655] 	mixer_test.go:250: Bad metric value for successful requests (200s): got 24.000000, want at most 23.000000
```

--- a/tests/e2e/tests/mixer/mixer_test.go
+++ b/tests/e2e/tests/mixer/mixer_test.go
@@ -446,7 +446,7 @@ func TestDenials(t *testing.T) {
 	}
 }
 
-func TestRateLimit(t *testing.T) {
+func TestMetricsAndRateLimitAndRulesAndBookinfo(t *testing.T) {
 	if err := replaceRouteRule(routeReviewsV3Rule); err != nil {
 		fatalf(t, "Could not create replace reviews routing rule: %v", err)
 	}
@@ -500,8 +500,8 @@ func TestRateLimit(t *testing.T) {
 	opts := fhttp.HTTPRunnerOptions{
 		RunnerOptions: periodic.RunnerOptions{
 			QPS:        10,
-			Exactly:    200,       // will make exactly 200 calls, so run for about 20 seconds
-			NumThreads: 5,         // get the same number of calls per connection (200/5=40)
+			Exactly:    300,       // will make exactly 200 calls, so run for about 30 seconds
+			NumThreads: 5,         // get the same number of calls per connection (300/5=60)
 			Out:        os.Stderr, // Only needed because of log capture issue
 		},
 		HTTPOptions: fhttp.HTTPOptions{
@@ -594,8 +594,8 @@ func TestRateLimit(t *testing.T) {
 		t.Logf("prometheus values for istio_request_count:\n%s", promDump(promAPI, "istio_request_count"))
 		errorf(t, "Bad metric value for successful requests (200s): got %f, want at least %f", got, want)
 	}
-
-	want200s = math.Ceil(want200s * 1.1) // timing is short, allow 10% extra for rounding errors and quota bucket size
+	// TODO: until https://github.com/istio/istio/issues/3028 is fixed, use 25% - should be only 5% or so
+	want200s = math.Ceil(want200s * 1.25)
 	if got > want200s {
 		t.Logf("prometheus values for istio_request_count:\n%s", promDump(promAPI, "istio_request_count"))
 		errorf(t, "Bad metric value for successful requests (200s): got %f, want at most %f", got, want200s)

